src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -76,47 +79,117 @@ public TensorFlowMapper(IHostEnvironment env, ISchema inputSchema, byte[] modelB
                 ctx.CheckAtModel(GetVersionInfo());
 
-                var numInputs = ctx.Reader.ReadInt32();
-                Contracts.CheckDecode(numInputs > 0);
+                var isFrozen = ctx.Reader.ReadInt32();
+                if (isFrozen==1)
+                {
+                    var numInputs = ctx.Reader.ReadInt32();
+                    Contracts.CheckDecode(numInputs > 0);
+
+                    string[] source = new string[numInputs];
+                    for (int j = 0; j < source.Length; j++)
+                        source[j] = ctx.LoadNonEmptyString();
+
+                    byte[] data = null;
+                    if (!ctx.TryLoadBinaryStream("TFModel", r => data = r.ReadByteArray()))
+                        throw env.ExceptDecode();
+
+                    var outputColName = ctx.LoadNonEmptyString();
+
+                    return new TensorFlowMapper(env, schema, data, source, outputColName);
+                }
+                else
+                {
+                    var numInputs = ctx.Reader.ReadInt32();
+                    Contracts.CheckDecode(numInputs > 0);
 
-                string[] source = new string[numInputs];
-                for (int j = 0; j < source.Length; j++)
-                    source[j] = ctx.LoadNonEmptyString();
+                    string[] source = new string[numInputs];
+                    for (int j = 0; j < source.Length; j++)
+                        source[j] = ctx.LoadNonEmptyString();
 
-                byte[] data = null;
-                if (!ctx.TryLoadBinaryStream("TFModel", r => data = r.ReadByteArray()))
-                    throw env.ExceptDecode();
+                    // Load model binary
+                    byte[] tfFilesBin = null;
+                    var load = ctx.TryLoadBinaryStream("TFSavedModel", br => tfFilesBin = br.ReadByteArray());
 
-                var outputColName = ctx.LoadNonEmptyString();
+                    var tempDirName = Path.GetFullPath(Path.Combine(Path.GetTempPath(), "_MLNET_TFTransform_" + Guid.NewGuid()));
+                    var tempDir = Directory.CreateDirectory(tempDirName);
+                    var tfZipFilePath = Path.Combine(tempDir.FullName, "tf_savedmodel.zip");
 
-                return new TensorFlowMapper(env, schema, data, source, outputColName);
+                    File.WriteAllBytes(tfZipFilePath, tfFilesBin);
+                    ZipFile.ExtractToDirectory(tfZipFilePath, Path.Combine(tempDir.FullName, "tf_savedmodel"));
+
+                    var outputColName = ctx.LoadNonEmptyString();
+
+                    return new TensorFlowMapper(env, schema, Path.Combine(tempDir.FullName, "tf_savedmodel"), source, outputColName);
+                }
             }
 
             public void Save(ModelSaveContext ctx)
             {
                 _host.AssertValue(ctx);
                 ctx.CheckAtModel();
                 ctx.SetVersionInfo(GetVersionInfo());
+                ctx.Writer.Write(_isFrozen ? 1 : 0);
```

Comment:
>Write [](start = 27, length = 5)

I believe you can write bool

Reply:
ctx.Writer.WriteBoolByte(IsFrozen);

---
In reply to: [216071878](https://github.com/dotnet/machinelearning/pull/853#discussion_r216071878) [](ancestors = 216071878)